src/hooks/useConnectOwnedVenues.tsx

```diff
@@ -0,0 +1,78 @@
+import { useMemo, useCallback } from "react";
+import { ReactHook } from "types/utility";
+import { AnyVenue } from "types/venues";
+
+import { WithId, withId } from "utils/id";
+import { ownedVenuesSelector } from "utils/selectors";
+
+import { useFirestoreConnect, isLoaded } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+
+export const useConnectOwnedVenues = (userId?: string) => {
+  useFirestoreConnect(
+    userId
+      ? {
+          collection: "venues",
+          where: [["owners", "array-contains", userId]],
+          storeAs: "ownedVenues",
+        }
+      : undefined
+  );
+};
+
+export interface useOwnedVenuesProps {
+  currentVenueId?: string;
+}
+
+export interface useOwnedVenuesData {
+  isLoaded: boolean;
+  isLoading: boolean;
+
+  currentVenue?: WithId<AnyVenue>;
+
+  ownedVenues: WithId<AnyVenue>[];
+  ownedVenuesIds: string[];
+
+  findVenueInOwnedVenues: (
+    searchedForVenueId: string
+  ) => WithId<AnyVenue> | undefined;
+}
+
+export const useOwnedVenues: ReactHook<
+  useOwnedVenuesProps,
+  useOwnedVenuesData
+> = ({ currentVenueId }): useOwnedVenuesData => {
+  const { userId } = useUser();
+  useConnectOwnedVenues(userId);
+
+  const venues = useSelector(ownedVenuesSelector);
+
+  const ownedVenues = useMemo(
+    () => Object.entries(venues ?? {}).map(([id, venue]) => withId(venue, id)),
+    [venues]
+  );
+
+  const findVenueInOwnedVenues = useCallback(
+    (searchedForVenueId: string): WithId<AnyVenue> | undefined =>
+      ownedVenues.find((venue) => venue.id === searchedForVenueId),
+    [ownedVenues]
+  );
```

Comment:
use ownedVenuesDataSelector here and get O(1) complexity instead of O(N)

Reply:
I should've caught this when I copied the code. Changed in 91bb8cbdd6f157e7c45f103389ebc034a5cdfa35